### PR TITLE
Bug 1219275 - Session is not kept between reboots. r=michielbdejong

### PR DESCRIPTION
**Pull request by ferjm**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=a78f473bf51f33af0a24374c6b1785f8c78db0af

File: tv_apps/smart-system/js/sync_manager.js [hunk 1]
```diff
@@ -501,6 +487,35 @@
...
     /** Helpers **/
 
+    observeSettings: function() {
+      var setSettingObserver = setting => {
+        return new Promise(resolve => {
+          SettingsListener.observe(setting, false, enabled => {
+            this.debug('Observed ' + setting + ' changed to ' + enabled);
+            enabled ? this.settings.set(setting, enabled)
+                    : this.settings.delete(setting);
+            resolve && resolve();
```
Comment (michielbdejong) [inline tv_apps/smart-system/js/sync_manager.js]: Why check for `resolve` to be defined here? It's defined on line 492, right?

File: tv_apps/smart-system/js/sync_manager.js [hunk 2]
```diff
@@ -501,6 +487,35 @@
...
+      ['sync.collections.bookmarks.enabled',
+       'sync.collections.history.enabled',
+       'sync.collections.passwords.enabled',
+       'sync.collections.bookmarks.readonly',
+       'sync.collections.history.readonly',
+       'sync.collections.passwords.readonly',
+       'sync.server.url',
+       'sync.scheduler.interval',
+       'sync.scheduler.wifionly',
+       'sync.fxa.audience'].forEach(setting => {
```
Comment (michielbdejong) [inline tv_apps/smart-system/js/sync_manager.js]: I usually do `Promise.all([ ... ].map(setting => setSettingObserver(setting)));` which avoids pushing to a second array, but I guess your syntax is easier to read, so you choose.

File: tv_apps/smart-system/js/sync_manager.js [hunk 3]
```diff
@@ -88,24 +88,6 @@
...
-           'sync.collections.history.enabled',
-           'sync.collections.passwords.enabled',
-           'sync.collections.bookmarks.readonly',
-           'sync.collections.history.readonly',
-           'sync.collections.passwords.readonly',
-           'sync.server.url',
-           'sync.scheduler.interval',
-           'sync.scheduler.wifionly',
-           'sync.fxa.audience'].forEach(setting => {
-            SettingsListener.observe(setting, false, enabled => {
```
Comment (michielbdejong) [inline tv_apps/smart-system/js/sync_manager.js]: Ah, so we were simply not waiting for it? ok, then this is a convincing fix.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=e2237137fd80fde95da2f59adff79f80d555e7cd

**Pull request merged**